core/sessions/user.go

```diff
@@ -18,13 +18,23 @@ import (
 type User struct {
 	Email             string
 	HashedPassword    string
+	Role              UserRole
 	CreatedAt         time.Time
 	TokenKey          null.String
 	TokenSalt         null.String
 	TokenHashedSecret null.String
 	UpdatedAt         time.Time
 }
 
+type UserRole string
+
+const (
+	UserRoleAdmin UserRole = "admin"
+	UserRoleEdit  UserRole = "edit"
+	UserRoleRun   UserRole = "run"
+	UserRoleView  UserRole = "view"
+)
```

Comment:
I wonder if it would be helpful to use an enum here? Then the order could be meaningful and allow you to do comparisions with `<`/`>` for sufficient auth level instead of multiple equality checks.
```suggestion
type UserRole int

const (
	UserRoleAdmin UserRole = iota
	UserRoleEdit
	UserRoleRun
	UserRoleView
	
	userRoleStrings = [...]string{UserRoleAdmin: "admin",UserRoleEdit: "edit", UserRoleRun: "run", UserRoleView: "view"}
)

func (r UserRole) String() { return userRoleStrings[r] }
```
Maybe not worth it - would complicate some other things.

Reply:
I think its a really cool idea, but I prefer the explicit simplicity + redundancy of just the static string, the behaviour is intuitive and easier to logically parse. In my opinion I think we should stick to the current implementation 🙂